lib/data/queries/country/getCountryWithApplicantCount.ts

```diff
@@ -2,12 +2,20 @@ import { Prisma } from "@prisma/client";
 import prisma from "../../../../prisma/client";
 
 //TODO add filter for status
-const getCountryWithApplicantCount = async (level?: string) => {
+const getCountryWithApplicantCount = async (
+  level?: string,
+  department?: string,
+) => {
   const levelFilter = level
     ? {
         equals: level,
       }
     : {};
+  const departmentFilter = department
+    ? {
+        equals: department,
+      }
+    : {};
```

Comment:
Here I would use a similar syntax as for the levels.

```suggestion
const departmentFilter = department
    ? {
        some: {
          id: {
            equals: department,
          },
        },
      }
    : {};
```